File: mediq-frontend/src/services/queueService.js
import api from './api';

export const queueService = {
  generateToken: async (data) => api.post('/queue/generate', data),
  updateTokenStatus: async (id, status, assignedDoctorId) =>
    api.patch(`/queue/${id}/status`, { status, assignedDoctorId }),
  assignDoctor: async (id, doctorId) => api.patch(`/queue/${id}/assign-doctor`, null, { params: { doctorId } }),
  getTokenById: async (id) => api.get(`/queue/${id}`),
  getNurseQueue: async (campId) => api.get(`/queue/nurse/${campId}`),
  getDoctorQueue: async (campId, doctorId) => api.get(`/queue/doctor/${campId}`, { params: { doctorId } }),
  getTokensByStatus: async (campId, status) => api.get(`/queue/status/${campId}`, { params: { status } }),
  getQueueDashboard: async (campId) => api.get(`/queue/dashboard/${campId}`),
  cancelToken: async (id) => api.delete(`/queue/${id}`),
};

export default queueService;

File: mediq-backend/src/main/java/com/mediq/service/QueueService.java
package com.mediq.service;

import com.mediq.constants.QueueStatus;
import com.mediq.dto.GenerateTokenRequest;
import com.mediq.dto.QueueDashboardResponse;
import com.mediq.dto.QueueTokenResponse;
import com.mediq.dto.UpdateQueueStatusRequest;

import java.util.List;

public interface QueueService {

    QueueTokenResponse generateToken(GenerateTokenRequest request);

    QueueTokenResponse updateTokenStatus(Long tokenId, UpdateQueueStatusRequest request);

    QueueTokenResponse assignDoctor(Long tokenId, Long doctorId);

    QueueTokenResponse getTokenById(Long tokenId);

    List<QueueTokenResponse> getNurseQueue(Long campId);

    List<QueueTokenResponse> getDoctorQueue(Long campId, Long doctorId);

    List<QueueTokenResponse> getTokensByStatus(Long campId, QueueStatus status);

    QueueDashboardResponse getQueueDashboard(Long campId);

    void cancelToken(Long tokenId);
}
